### ISSUE TITLE
Create a "table_info" admin table

### ISSUE THREAD

**Issue by deontologician** [rethinkdb/rethinkdb#4522]
Currently, doc counts and a list of secondary indexes needs to be obtained directly on a table through `.info()` (or `.indexList()`). This information wasn't added to `table_status` because it's slow to gather, and since we can't return a value until the slowest information returns, it would be an unnecessary drag on `table_status`'s performance.

I still think there's value in have a `table_info` table because `r.db(x).table(y).info()` will succeed only if the table exists, whereas `table_status` and `table_config` always exist and can return error documents or incomplete documents with whatever non-authoritative information is available.

This makes admin queries easier, and makes thinks slightly more symmetrical with `.status()` and `.config()`. The major downside is that this would be a slow table, and `.info()` indicates that it's an operation rather than pretending it's data that's already hanging around (as a table implies). I think if we mention in the docs that the table is potentially very slow it would be ok.

**Comment by danielmewes** [comment]
I think it would be acceptable to have a slow table if it makes things significantly easier.

Are you able to work around this for the time being?

**Comment by deontologician** [comment]
Unfortunately, `r.table(x).info().default({..})` won't work where x is an ambiguous table name or doesn't exist. So it's impossible to compose into a larger query. The only work-around is to split out the queries that use `.info` and `.indexList` into their own queries because there is no way to recover from that error in reql currently.

**Comment by danielmewes** [comment]
In the web UI, is this primarily relevant for getting secondary indexes or for getting document counts?

Secondary indexes are now stored in the raft state, so I think we could include a list with their names in `table_status` without significant performance drawbacks.

**Comment by deontologician** [comment]
In terms of work that needs to be done immediately, secondary indexes are needed for the dashboard, so having it in table_status would help. Other places in the webui use document counts, but those have already been split into multiple queries previously.

**Comment by danielmewes** [comment]
Just talked to @VeXocide . We can easily put a list of secondary indexes into `table_config`. For now we will keep it read-only and simply list the index names like we do in `table.info()`. Assigning @VeXocide

**Comment by deontologician** [comment]
Could we fork it to another issue? This issue as described would be a better long term change

**Comment by danielmewes** [comment]
Yeah I'm not sure if we still need the `table_info` table after that. Let me create a separate issue for now and move this to backlog then.

**Comment by danielmewes** [comment]
-> https://github.com/rethinkdb/rethinkdb/issues/4525
